ThreeBotPackages/threefold/tfgrid_workloads/docs/reservation.md

```diff
@@ -0,0 +1,94 @@
+# Reservation
+
+The reservation model consists of 2 parts: the reservation data, and the
+reservation state. The data is is created entirely when the reservation is created,
+and is immutable after this. The state is mutable, and after the reservation has been
+created, this will be updated continuously by both the customer threebot and the farmer
+threebot (and possibly other threebots who will sign to trigger actions).
+
+## Reservation data
+
+The reservation data is composed of the individual workloads to be deployed, information
+about who can sign for the actual deployment and deletion of the workloads, and expiry
+times for the reservation. All fields in the data object are immutable after the
+reservation is created (i.e. the customer signs the data). Modifications afterwards
+will cause the customer signature to become invalid.
+
+## Reservation state
+
+The reservation state is updated throughout the lifetime of the reservation. It also
+contains the signatures needed to have the farmer threebot take action. In order for
+the farmer threebot to start provisioning the workloads, or delete the workloads,
+the `signing_request_provision` and `signing_request_delete`, respectively, need to be filled with
+valid signatures. A valid signature is a signature for the reservation data, with a
+private key owned by one of the threebots listed in the reservation data (in the
+`signatures_provision` and `signatures_delete` fields). These fields also define
+the minimum amount of signatures required. For example, a signature request for
+provisioning might list 3 threebot ids which can sign, but only specify a `quorum_min`
+of 2. As such, only 2 out of the 3 listed threebot ids would need to sign before the
+farmer threebot is allowed to deploy the reserved workloads.
+
+### signature validity
```

Comment:
we should also add a full description of what kind of signature we use. Algorigthm, size of keys...